crates/nu_plugin_polars/src/dataframe/command/core/save/mod.rs

```
@@ -223,6 +230,28 @@ fn command(
     Ok(PipelineData::empty())
 }
 
+fn check_writing_into_source_file(
+    metadata: Option<&PipelineMetadata>,
+    dest: &Spanned<PathBuf>,
+) -> Result<(), ShellError> {
+    let Some(DataSource::FilePath(source)) = metadata.map(|meta| &meta.data_source) else {
+        return Ok(());
+    };
+
```

Can be changed to only guard `.parquet` files, since this panic doesn't happen on `.csv` ones for example, but I'm not super familiar with polars to make this call, so for consistency I'm including all extensions

---

I am not sure if despite not panicking the reader may read an empty csv as the writer starts opening the file and thus corrupting it.

---

Just tested and it indeed is silently corrupting the csv, so I guess this is even better because it will prevent silent errors